pkg/parsers/azure/steps.go

```diff
@@ -0,0 +1,136 @@
+package azure
+
+import (
+	"regexp"
+	"strings"
+
+	azureModels "github.com/argonsecurity/pipeline-parser/pkg/loaders/azure/models"
+	loaderUtils "github.com/argonsecurity/pipeline-parser/pkg/loaders/utils"
+	"github.com/argonsecurity/pipeline-parser/pkg/models"
+	"github.com/argonsecurity/pipeline-parser/pkg/utils"
+)
+
+var (
+	azureTaskNameRegex = regexp.MustCompile(`(.+?)(?:@(.+)|$)`)
+)
+
+func parseSteps(steps *azureModels.Steps) []*models.Step {
+	if steps == nil {
+		return nil
+	}
+
+	return utils.Map(*steps, parseStep)
+}
+
+func parseStep(step azureModels.Step) *models.Step {
+	parsedStep := &models.Step{
+		Name:                 &step.DisplayName,
+		EnvironmentVariables: parseEnvironmentVariablesRef(step.Env),
+		FileReference:        step.FileReference,
+	}
+
+	if step.ContinueOnError != nil {
+		parsedStep.FailsPipeline = utils.GetPtr(!*step.ContinueOnError)
```

Comment:
Is `ContinueOnError` never nil? if it can be nil then `*(nil pointer)` throws panic